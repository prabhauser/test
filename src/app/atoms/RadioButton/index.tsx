import { Radio, FormHelperText, RadioGroupProps } from '@material-ui/core';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel, { FormControlLabelProps } from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import FormLabel from '@material-ui/core/FormLabel';
import { makeStyles } from '@material-ui/core/styles';
import * as defaultTheme from '../../../utilities/theme/theme';

type ComponentProps = RadioGroupProps & {
    options: any;
    label?: string;
    labelPlacement?: FormControlLabelProps['labelPlacement'];
    disabled?: boolean;
    inputRef?: FormControlLabelProps['inputRef'];
    helperText?: string;
};

const useStyles = makeStyles(() => ({
    root: {
        '& .Mui-checked': {
            color: `${defaultTheme.colorCodes.supportRed}`
        },
        '& .MuiFormHelperText-root': {
            color: `${defaultTheme.colorCodes.error}`,
            fontSize: '12px',
            lineHeight: '14px',
            marginTop: '10px'
        }
    }
}));

const RadioButton = (props: ComponentProps) => {
    const { options, label, labelPlacement, disabled, inputRef, helperText, ...rest } = props;
    const classes = useStyles();
    return (
        <FormControl component="fieldset" className={`${classes.root}`}>
            {label && <FormLabel>{label}</FormLabel>}
            <RadioGroup {...rest}>
                {options?.map((option: any, index: number) => {
                    const radioKey = index + 1;
                    return (
                        <FormControlLabel
                            key={radioKey}
                            value={option?.value}
                            control={<Radio />}
                            label={option?.label}
                            labelPlacement={labelPlacement || 'end'}
                            disabled={option?.disabled || disabled}
                            inputRef={inputRef}
                        />
                    );
                })}
            </RadioGroup>
            {helperText && <FormHelperText>{helperText}</FormHelperText>}
        </FormControl>
    );
};

export default RadioButton;
