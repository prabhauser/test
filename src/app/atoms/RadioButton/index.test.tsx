import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import RadioButton from './index';

describe('Radio Button Component', () => {
    it('Radio Button Screen snapshot', () => {
        const component = shallow(<RadioButton options={['first', 'second']} />);
        expect(component).toMatchSnapshot();
    });

    it('Radio button label should be in DOM', () => {
        const test1 = { title: 'Radio Label' };
        const test2 = {
            title: 'Test2'
        };

        const { getByText, queryByText } = render(<RadioButton options={['first', 'second']} label={test1.title} />);

        expect(getByText(test1.title)).toBeInTheDocument();
        expect(queryByText(test2.title)).toBeNull();
    });
});
