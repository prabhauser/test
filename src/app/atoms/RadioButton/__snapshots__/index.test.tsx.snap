// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button Component Radio Button Screen snapshot 1`] = `
<WithStyles(ForwardRef(FormControl))
  className="makeStyles-root-1"
  component="fieldset"
>
  <ForwardRef(RadioGroup)>
    <WithStyles(ForwardRef(FormControlLabel))
      control={<WithStyles(ForwardRef(Radio)) />}
      key="1"
      labelPlacement="end"
    />
    <WithStyles(ForwardRef(FormControlLabel))
      control={<WithStyles(ForwardRef(Radio)) />}
      key="2"
      labelPlacement="end"
    />
  </ForwardRef(RadioGroup)>
</WithStyles(ForwardRef(FormControl))>
`;
